qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```
@@ -0,0 +1,709 @@
+
+    def _set_exclusive_slot_close(self, val: str) -> None:
+        slots = ['allslots', *self._get_slot_name()]
+        if val in slots:
+            raise InvalidValue("Slots cannot be exclusively closed.")
+        if val == "":
+            raise InvalidValue('Argument cannot be an empty string.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels or channel '
+                               'ranges and associated backplane relays.')
+        self.write(f"channel.exclusiveslotclose('{val}')")
+
+    def _get_channel_connect_rule(self) -> str:
+        connect_rule = {1: 'BREAK_BEFORE_MAKE',
+                        2: 'MAKE_BEFORE_BREAK',
+                        0: 'OFF'}
+        rule = self.ask('channel.connectrule')
+        return connect_rule[int(float(rule))]
+
+    def _set_channel_connect_rule(self, val: str) -> None:
+        self.write(f'channel.connectrule = channel.{val}')
+
+    def _get_gpib_status(self) -> str:
+        return self.ask('comm.gpib.enable')
+
+    def _set_gpib_status(self, val: Union[str, bool]) -> None:
+        self.write(f'comm.gpib.enable = {val}')
+
+    def _get_lan_status(self) -> str:
+        return self.ask('comm.lan.enable')
+
+    def _set_lan_status(self, val: Union[str, bool]) -> None:
+        self.write(f'comm.lan.enable = {val}')
+
+    def _get_gpib_address(self) -> int:
+        return int(float(self.ask('gpib.address')))
+
+    def _set_gpib_address(self, val: int) -> None:
+        self.write(f'gpib.address = {val}')
+
+    def get_closed_channels(self, val: str) -> Union[str, None]:
+        """
+        Queries for the closed channels.
+
+        Args:
+            val: A string representing the channels,
+                backplane relays or channel patterns to be queried.
+        """
+        if val == "":
+            raise InvalidValue('Argument cannot be an empty string.')
```

isn't this captured by the `_validator`?

---

No because this is an edge case where in fact the instrument returns a silent error code, thus implemented individually here.